### PR TITLE
Issue #20428: Don't throw UnsupportedOperationException

### LINKED ISSUE #20428: Servlet 6.0: Remove the restriction on programatically added listeners calling some ServletContext getter methods.
Spec issue: `https://github.com/eclipse-ee4j/servlet-api/issues/416`

See "Appendix A: Change Log" here: https://github.com/eclipse-ee4j/servlet-api/blob/master/spec/src/main/asciidoc/servlet-spec-body.adoc

Some ServletContext getter methods no longer throw an `UnsupportedOperationExceptio` in the following condition (this was removed from the getter javadocs:

>    @throws UnsupportedOperationException if this ServletContext was passed to the
      {@link ServletContextListener#contextInitialized} method of a {@link ServletContextListener} that was neither
      declared in <code>web.xml</code> or <code>web-fragment.xml</code>, nor annotated with
      {@link jakarta.servlet.annotation.WebListener}

This should be the complete list of methods that need to be updated for Servlet 6.0:

- getEffectiveMajorVersion -> new requirement in servlet-3.1
- getEffectiveMinorVersion -> new requirement in servlet-3.1
- getDefaultSessionTrackingModes
- getEffectiveSessionTrackingModes
- getJspConfigDescriptor
- getClassLoader -> new requirement in servlet-3.1
- getVirtualServerName -> new requirement in server-3.1, new method in servlet-3.1
- getSessionTimeout -> new method in servlet-4.0
- getRequestCharacterEncoding -> new method in servlet-4.0
- getResponseCharacterEncoding -> new method in servlet-4.0

1) Validate that the above is a complete list by reviewing the [ServletContext](https://jakarta.ee/specifications/servlet/6.0/apidocs/jakarta.servlet/jakarta/servlet/servletcontext) JavaDoc.
2) https://github.com/OpenLiberty/open-liberty/blob/2fd4a880754c37a988c5ed9ac4f1ea5988e465d6/dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/facade/ServletContextFacade.java will need to be updated for Servlet 6.0 as well as https://github.com/OpenLiberty/open-liberty/blob/2fd4a880754c37a988c5ed9ac4f1ea5988e465d6/dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/webapp/WebApp.java

### PR DESCRIPTION
**Pull request by pnicolucci**
fixes #20428

I needed to add the JSP feature to the new test server since we get an NPE during EE10 executions: https://github.com/OpenLiberty/open-liberty/issues/21820

### CODE REVIEW

**Comment by wraschke** [comment]
#libby

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby #build

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/publish/servers/servlet31_ServletContext_UnsupportedOperationExceptionr/bootstrap.properties [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+###############################################################################
+bootstrap.include=../testports.properties
+osgi.console=7777
+com.ibm.ws.logging.trace.specification=*=info
```
Comment (pmd1nh) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/publish/servers/servlet31_ServletContext_UnsupportedOperationExceptionr/bootstrap.properties]: perhaps adding WC traces in the trace.specification for debugging if needed.

Comment (pmd1nh) [reply dev/com.ibm.ws.webcontainer.servlet.3.1_fat/publish/servers/servlet31_ServletContext_UnsupportedOperationExceptionr/bootstrap.properties]: `*=info=enabled:com.ibm.ws.webcontainer*=all:com.ibm.wsspi.webcontainer*=all`

File: dev/com.ibm.ws.webcontainer.servlet.4.0_fat/publish/servers/servlet40_ServletContext_UnsupportedOperationExceptionr/bootstrap.properties [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+###############################################################################
+bootstrap.include=../testports.properties
+osgi.console=7777
+com.ibm.ws.logging.trace.specification=*=info:com.ibm.ws.webcontainer*=all:com.ibm.wsspi.webcontainer*=all:HttpTransport=all:HTTPChannel=all:TCPChannel=all:GenericBNF=all:com.ibm.ws.jsp*=all:com.ibm.ws.session.*=all:com.ibm.ws.runtime.update.internal*=all:ChannelFramework=all
```
Comment (pmd1nh) [inline dev/com.ibm.ws.webcontainer.servlet.4.0_fat/publish/servers/servlet40_ServletContext_UnsupportedOperationExceptionr/bootstrap.properties]: remove com.ibm.ws.jsp* session.* and runtime.update.internal*.  We don't need these traces for these tests.

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby

**Review by pmd1nh — APPROVED** [review]

**Review by volosied — APPROVED** [review]

**Pull request merged**